### PR TITLE
Fix ambiguous Utf8JsonReader Get/TryGet method documentation

### PR DESCRIPTION
**Pull request by Wildenhaus**
Fixes issue #37722

### CODE REVIEW

File: src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.TryGet.cs [hunk 1]
```diff
@@ -40,7 +40,7 @@ public string GetString()
         }
 
         /// <summary>
-        /// Reads the next JSON token value from the source as a <see cref="bool"/>.
+        /// Parses the current JSON token value as a <see cref="bool"/>.
```
Comment (ahsonkhan) [inline src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.TryGet.cs]: nit: Maybe keep the `from the source` phrase?

```C#
[Fact]
public static void ChangeSource()
{
    string jsonString = "12345";
    byte[] dataUtf8 = Encoding.UTF8.GetBytes(jsonString);

    var json = new Utf8JsonReader(dataUtf8, isFinalBlock: true, state: default);
    Assert.True(json.Read());
    Assert.Equal(JsonTokenType.Number, json.TokenType);
    Assert.Equal(dataUtf8.Length, json.BytesConsumed);
    Assert.Equal(json.BytesConsumed, json.CurrentState.BytesConsumed);

    Assert.Equal(12345, json.GetInt32());

    dataUtf8[4] = (byte)'6'; // User changed the source data
    Assert.Equal(12346, json.GetInt32());
}
```

File: src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.TryGet.cs [hunk 2]
```diff
@@ -11,7 +11,7 @@ namespace System.Text.Json
     public ref partial struct Utf8JsonReader
     {
         /// <summary>
-        /// Reads the next JSON token value from the source, unescaped, and transcoded as a <see cref="string"/>.
+        /// Parses the current JSON token value unescaped and transcoded as a <see cref="string"/>.
```
Comment (ahsonkhan) [inline src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.TryGet.cs]: nit: Same here/elsewhere, I would keep `from the source`.

Also nit: Imo, keeping the comma helps with clarity while reading it.

**Review by ahsonkhan — APPROVED** [review]
Otherwise, LGTM. Thanks for catching and fixing this.

**Comment by Wildenhaus** [comment]
Pushed the requested changes.

**Review by ahsonkhan — APPROVED** [review]

**Comment by ahsonkhan** [comment]
@Wildenhaus, we added a `GetComment()` API. Can you please resolve the merge conflicts and then we can check this in. Thanks!

**Comment by Wildenhaus** [comment]
Fixed the merge conflicts. Do you want me to rebase?

**Comment by ahsonkhan** [comment]
> Do you want me to rebase?

Nope, that isn't necessary here. Thanks.

**Pull request merged**